pkg/cmd/bootstrap.go

```diff
@@ -168,34 +176,82 @@ func initiateInteractiveMode(io *BootstrapParameters, client *utility.Client) er
 		identifier := factory.NewDriverIdentifier(factory.Mapping(host, io.PrivateRepoDriver))
 		factory.DefaultIdentifier = identifier
 	}
-	if ui.UseInternalRegistry() {
-		io.ImageRepo = ui.EnterImageRepoInternalRegistry()
-	} else {
-		io.ImageRepo = ui.EnterImageRepoExternalRepository()
-		io.DockerConfigJSONFilename = ui.EnterDockercfg()
+	if cmd.Flag(imageRepoFlag).Changed {
+		isInternalRegistry, _, err := imagerepo.ValidateImageRepo(io.ImageRepo)
+		if !isInternalRegistry {
+			if !cmd.Flag("dockercfgjson").Changed {
+				log.Progressf("The supplied image repository has been detected as an external repository.")
+				io.DockerConfigJSONFilename = ui.EnterDockercfg()
+			}
+		}
+		if err != nil {
+			return err
+		}
+	} else if promptForAll {
+		if ui.UseInternalRegistry() {
+			io.ImageRepo = ui.EnterImageRepoInternalRegistry()
+		} else {
+			io.ImageRepo = ui.EnterImageRepoExternalRepository()
+			io.DockerConfigJSONFilename = ui.EnterDockercfg()
+		}
+	}
+	if promptForAll {
+		io.GitOpsWebhookSecret = ui.EnterGitWebhookSecret(io.GitOpsRepoURL)
+	}
+	if !cmd.Flag(serviceRepoURLFlag).Changed {
+		io.ServiceRepoURL = ui.EnterServiceRepoURL()
+	}
+	io.ServiceRepoURL = utility.AddGitSuffixIfNecessary(io.ServiceRepoURL)
+	if promptForAll {
+		io.ServiceWebhookSecret = ui.EnterGitWebhookSecret(io.ServiceRepoURL)
 	}
-	io.GitOpsWebhookSecret = ui.EnterGitWebhookSecret(io.GitOpsRepoURL)
-	io.ServiceRepoURL = utility.AddGitSuffixIfNecessary(ui.EnterServiceRepoURL())
-	io.ServiceWebhookSecret = ui.EnterGitWebhookSecret(io.ServiceRepoURL)
 	secret, err := accesstoken.GetAccessToken(io.ServiceRepoURL)
 	if err != nil && err != keyring.ErrNotFound {
 		return err
 	}
-	if secret == "" {
-		io.GitHostAccessToken = ui.EnterGitHostAccessToken(io.ServiceRepoURL)
-		io.SaveTokenKeyRing = ui.UseKeyringRingSvc()
+	if secret == "" { // We must prompt for the token
+		if !cmd.Flag(statustracker.CommitStatusTrackerSecret).Changed {
```

Comment:
Again, given that there's no default for the CommitStatusTrackerSecret, we can simplify this to just...`if io.GitHostAccessToken == "" {`

Reply:
Same rationale as in the previous comment.  Using "" ties the default value here when it's defined in line 383 and on.

Reply:
Right, but there's no sensible default value for the commit status tracker secret?

Nor for the service URL or the GitOps URL?